Dockerfile

```diff
@@ -1,151 +1,121 @@
 
-RUN rsync -a --delete node_modules rootfs-overlay /build
+RUN npm ci
 
-RUN [ "cross-build-end" ]
+COPY webpack.config.js fix-jsonstream.js hardcode-migrations.js tsconfig.json tsconfig.release.json ./
+COPY src ./src
+COPY test ./test
+COPY typings ./typings
 
-##############################################################################
+RUN npm run test-nolint \
+	&& npm run build
 
-# Minimal runtime image
-FROM balena/$ARCH-supervisor-base:v1.4.7
+# Run the production install here, to avoid the npm dependency on
+# the later stage
+RUN npm ci --production --no-optional --unsafe-perm \
+	&& npm cache clean --force \
+	# For some reason this doesn't get cleared with the other
+	# cache
+	&& rm -rf node_modules/.cache \
+	# Remove various uneeded filetypes in order to reduce space
+	# We also remove the spurious node.dtps, see https://github.com/mapbox/node-sqlite3/issues/861
+		&& find . -path '*/coverage/*' -o -path '*/test/*' -o -path '*/.nyc_output/*' \
+			-o -name '*.tar.*'      -o -name '*.in'     -o -name '*.cc' \
+			-o -name '*.c'          -o -name '*.coffee' -o -name '*.eslintrc' \
+			-o -name '*.h'          -o -name '*.html'   -o -name '*.markdown' \
+			-o -name '*.md'         -o -name '*.patch'  -o -name '*.png' \
+			-o -name '*.yml'        -o -name "*.ts" \
+			-delete \
+			&& find . -type f -path '*/node_modules/sqlite3/deps*' -delete \
+		&& find . -type f -path '*/node_modules/knex/build*' -delete \
+		&& rm -rf node_modules/sqlite3/node.dtps
+
+
+# RUN ["cross-build-end"]
+
+FROM balenalib/$ARCH-alpine-supervisor-base:3.11
+
+# RUN ["cross-build-start"]
+
+RUN apk add --no-cache \
+	ca-certificates \
+	kmod \
+	iptables \
+	ip6tables \
+	rsync \
+	avahi \
+	dbus \
+	libstdc++
 
 WORKDIR /usr/src/app
 
-COPY --from=node-build /usr/src/app/dist ./dist
-COPY --from=node-deps /build/node_modules ./node_modules
-COPY --from=node-deps /build/rootfs-overlay/ /
+COPY --from=BUILD /usr/local/bin/node /usr/local/bin/node
+COPY --from=BUILD /usr/src/app/dist ./dist
+COPY --from=BUILD /usr/src/app/package.json ./
+COPY --from=BUILD /usr/src/app/node_modules ./node_modules
 
-# Remove default nproc limit for Avahi for it to work in-container
-COPY avahi-daemon.conf /etc/avahi/avahi-daemon.conf
+COPY entry.sh .
 
-VOLUME /data
+RUN mkdir -p rootfs-overlay && \
+	(([ ! -d rootfs-overlay/lib64 ] && ln -s /lib rootfs-overlay/lib64) || true)
 
 ARG ARCH
 ARG VERSION=master
 ARG DEFAULT_MIXPANEL_TOKEN=bananasbananas
 ENV CONFIG_MOUNT_POINT=/boot/config.json \
 	LED_FILE=/dev/null \
-	SUPERVISOR_IMAGE=resin/$ARCH-supervisor \
+	SUPERVISOR_IMAGE=balena/$ARCH-supervisor \
 	VERSION=$VERSION \
 	DEFAULT_MIXPANEL_TOKEN=$DEFAULT_MIXPANEL_TOKEN
+COPY avahi-daemon.conf /etc/avahi/avahi-daemon.conf
 
+VOLUME /data
 HEALTHCHECK --interval=5m --start-period=1m --timeout=30s --retries=3 \
-	CMD wget -qO- http://127.0.0.1:${LISTEN_PORT:-48484}/v1/healthy || exit 1
+	CMD wget http://127.0.0.1:${LISTEN_PORT:-48484}/v1/healthy -O - -q
```

Comment:
why no more `|| exit 1`?

Reply:
It's not needed, wget will `exit 1` if there's no reply, or bad response.

Reply:
I mean the `||` kind of implies it already exited with `1` (or some other failure code, but I'd expect they'd all be handled the same as `1` anyway)